### PR TITLE
Integration tests with docker

### PR DESCRIPTION
**Pull request by kocevaIv**
In this pull request several things are added for the project:

- docker-compose.yml file is added for running integration tests with mongo in docker

- DockerComposeRule is created 

- .travis.yml file is created for running integration tests and unit tests with travis

- Maven plugin failsafe is added

### CODE REVIEW

File: api/src/test/java/com/brandwatch/ivanatwitterapp/api/integrationtests/MentionIT.java [hunk 1]
```diff
@@ -45,6 +47,7 @@
             Month.SEPTEMBER,
             23,
             13, 30);
+    private Logger log = LoggerFactory.getLogger(MentionIT.class);
```
Comment (amcmurray-bw) [inline api/src/test/java/com/brandwatch/ivanatwitterapp/api/integrationtests/MentionIT.java]: This logger could be immutable.

File: api/src/test/java/com/brandwatch/ivanatwitterapp/api/integrationtests/MongoTestConfig.java [hunk 2]
```diff
@@ -1,13 +1,43 @@
...
-public class MongoTestConfig {
+public class MongoTestConfig extends AbstractMongoConfiguration {
 
     @Bean(initMethod = "populateTestDatabase")
     public DatabaseFixtures databaseManager() {
         return new DatabaseFixtures();
     }
+
+    @Bean(initMethod = "before", destroyMethod = "after")
+    public DockerComposeRule dockerComposeRule() throws IOException, InterruptedException {
```
Comment (amcmurray-bw) [inline api/src/test/java/com/brandwatch/ivanatwitterapp/api/integrationtests/MongoTestConfig.java]: You might want to consider writing logs from the tests somewhere. You can configure the `DockerComposeRule` be configured to do this, you can read about it [here](https://github.com/palantir/docker-compose-rule#collecting-logs).

File: pom.xml [hunk 3]
```diff
@@ -1,36 +1,47 @@
...
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-	<modelVersion>4.0.0</modelVersion>
-
-	<groupId>twitterapp</groupId>
-	<artifactId>app</artifactId>
-	<version>0.0.1-SNAPSHOT</version>
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <build>
```
Comment (ivinskyi) [inline pom.xml]: Usually we place `<build>` section somewhere in the middle or at the end of pom.xml, all the other tags, that are currently in this file, are more informative and small and should stay at the top.

File: api/src/test/java/com/brandwatch/ivanatwitterapp/api/integrationtests/MongoTestConfig.java [hunk 4]
```diff
@@ -1,13 +1,48 @@
...
+                .build();
+        return dockerComposeRule;
+    }
+
+    @Override
+    protected String getDatabaseName() {
+        return "testTweets";
+    }
+
+
```
Comment (ivinskyi) [inline api/src/test/java/com/brandwatch/ivanatwitterapp/api/integrationtests/MongoTestConfig.java]: Extra blank line

File: api/src/test/java/com/brandwatch/ivanatwitterapp/api/integrationtests/MentionIT.java [hunk 5]
```diff
@@ -57,10 +57,12 @@ public void testReadMentions() throws IOException {
                 + "&startDate=" + START_DATE
                 + "&endDate=" + END_DATE;
         String response = testRestTemplate.getForObject(fullUrl, String.class);
-        List<Mention> actualMentions = objectMapper.readValue(response, new TypeReference<List<Mention>>() {});
+        List<Mention> actualMentions = objectMapper.readValue(response, new TypeReference<List<Mention>>() {
+        });
```
Comment (ahmedHathout) [inline api/src/test/java/com/brandwatch/ivanatwitterapp/api/integrationtests/MentionIT.java]: I think it should be on the same line as it was before

**Pull request merged**